src/server/world.zig

```diff
@@ -528,6 +531,18 @@ pub const ServerWorld = struct { // MARK: ServerWorld
 		try files.writeZon(try std.fmt.bufPrint(&buf, "saves/{s}/palette.zig.zon", .{name}), self.blockPalette.save(arenaAllocator));
 		try files.writeZon(try std.fmt.bufPrint(&buf, "saves/{s}/biome_palette.zig.zon", .{name}), self.biomePalette.save(arenaAllocator));
 
+		var gamerules = files.readToZon(arenaAllocator, try std.fmt.bufPrint(&buf, "saves/{s}/gamerules.zig.zon", .{name})) catch blk: {
+			const gameruleZon = ZonElement.initObject(arenaAllocator);
+
+			try files.writeZon(try std.fmt.bufPrint(&buf, "saves/{s}/gamerules.zig.zon", .{name}), gameruleZon);
+
+			break :blk gameruleZon;
+		};
+		defer gamerules.deinit(arenaAllocator);
```

Comment:
We don't need to deinit, since this is an arenaAllocator which gets freed at the end of the scope anyways.